### DISCUSSION TITLE
Execution Flow

### DISCUSSION THREAD

**Discussion by SimsonMoses** [keycloak/keycloak#33688] [Q&A]
I have using the SPI keycloak to customize the flow while restriction two action has been set VERIFY_EMAIL & UPDATE_PASSWORD. verify email has been sent. Once email verify action done redirect automatically to update password page. this flow working on the same browser where registration done but while link opened on the other browser or private window only email verified, update password page does not shown why??

how to resolve the issue??

Can any one help me to sort out it?

**Comment by codespearhead** [comment]
Keycloak uses cookies to track the session state, which aren't available in other browsers or private windows.

Upgrading to Keycloak 26 [1] might solve this issue.

[1] https://github.com/keycloak/keycloak/blob/98a6ddead0d7121470f5d17be635e1f4dbf3058e/docs/documentation/release_notes/topics/26_0_0.adoc?plain=1#L28
